test/hotspot/jtreg/testlibrary/ctw/src/sun/hotspot/tools/ctw/Compiler.java

```diff
@@ -79,26 +81,53 @@ public static void compileClass(Class<?> aClass, long id, Executor executor) {
             preloadClasses(aClass.getName(), id, constantPool);
         }
 
-        // Make sure the class is initialized.
-        UNSAFE.ensureClassInitialized(aClass);
+        // Attempt to initialize the class. If initialization is not possible
+        // due to NCDFE, accept this, and try compile anyway.
+        try {
+            UNSAFE.ensureClassInitialized(aClass);
+        } catch (NoClassDefFoundError e) {
+            CompileTheWorld.OUT.println(String.format("[%d]\t%s\tNOTE unable to init class : %s",
+                                 id, aClass.getName(), e));
+        }
         compileClinit(aClass, id);
 
+        // Getting constructor/methods with unresolvable signatures would fail with NCDFE.
+        // Try to get as much as possible, and compile everything else.
+        // TODO: Would be good to have a Whitebox method that returns the subset of resolvable
+        // constructors/methods without throwing NCDFE. This would extend the testing scope.
+        Constructor[] constructors = new Constructor[0];
+        Method[] methods = new Method[0];
+
+        try {
+            constructors = aClass.getDeclaredConstructors();
+        } catch (NoClassDefFoundError e) {
+            CompileTheWorld.OUT.println(String.format("[%d]\t%s\tNOTE unable to get constructors : %s",
```

Comment:
Nitpick really but why not call `CompileTheWorld.OUT.printf(...` instead of `CompileTheWorld.OUT.println(String.format(...`?